provider/baked/src/zerotrie.rs

```diff
@@ -39,19 +42,38 @@ pub(crate) fn bake(
 
     (
         quote! {
-            icu_provider_baked::zerotrie::Data<#marker_bake> = icu_provider_baked::zerotrie::Data {
-                trie: icu_provider_baked:: #baked_trie,
-                values: &[#(#bakes,)*],
+            // Safety invariant upheld: see above
+            icu_provider_baked::zerotrie::Data<#marker_bake> = {
+                const TRIE: icu_provider_baked::zerotrie::ZeroTrieSimpleAscii<&'static [u8]> = icu_provider_baked:: #baked_trie;
+                const VALUES: &'static [<#marker_bake as icu_provider_baked::zerotrie::DynamicDataMarker>::DataStruct] = &[#(#bakes,)*];
+                unsafe {
+                    icu_provider_baked::zerotrie::Data::from_trie_and_values_unchecked(TRIE, VALUES)
+                }
```

Comment:
can you try inlining these two consts? Try `as_slice()` if it complains about `&[T; N]` vs `&[T]`.

Reply:
it's a massive one-liner currently. This change makes it into a massive N+4 liner